azure-pipelines.yml

```diff
@@ -121,6 +121,37 @@ jobs:
         DISTRIB: 'conda-pypy3'
         LOCK_FILE: './build_tools/azure/pypy3_linux-64_conda.lock'
 
+
+- job: Linux_Nightly_Pyodide
+  pool:
+    vmImage: ubuntu-22.04
+  variables:
+    # Need to match Python version and Emscripten version for the correct
+    # Pyodide version. For Pyodide version 0.23.2, see
+    # https://github.com/pyodide/pyodide/blob/0.23.2/Makefile.envs
+    PYODIDE_VERSION: '0.23.2'
+    EMSCRIPTEN_VERSION: '3.1.32'
+    PYTHON_VERSION: '3.11.2'
+
+  dependsOn: [git_commit, linting]
+  condition: |
+    and(
+      succeeded(),
+      not(contains(dependencies['git_commit']['outputs']['commit.message'], '[ci skip]')),
+      or(eq(variables['Build.Reason'], 'Schedule'),
+         contains(dependencies['git_commit']['outputs']['commit.message'], '[pyodide]'
+        )
+      )
+    )
+  steps:
```

Comment:
Can the steps be moved into it's own `build_tools/azure/pyodide.yml` file?

Reply:
I don't really understand the need to introduce a new template if it is meant to be instantiated only once.

Reply:
Maybe to keep things simpler in `azure-pipelines.yml`? `azure-pipelines.yml` is the high-level overview and then the template have all the details.

To be honest, I am happy to do either, let me know!

Reply:
> Maybe to keep things simpler in azure-pipelines.yml? 

That was my original idea. In any case, I do not think this is a blocker and we can merge as is.